### ISSUE TITLE
No DateTime comparison which compares both date and time? 

### ISSUE THREAD

**Issue by brado** [bitwalker/timex#148]
I've spent several hours testing out the DateTime comparison and diff functions. I also read through some of the source code, and if I have understood it correctly, the compare function does not compare the date *and* the time. In other words, two DateTimes with the same date but different times will compare as being equal, and if granularity (such as seconds) is selected, it compares only that specific part -- seconds -- it does not compare date *and* time down to the seconds. 

Have I understood this correctly, and what is the intent of the compare function. Is there a single function to compare two DateTimes and have both its date and time components compared to determine which is greater / lesser / equal?

### Steps to reproduce

- Compare two DateTime structs.

### Description of issue

- I would expect a comparison that takes into account date *and* time, with the level of granularity specifying the level down to which the date *and* time were compared, not an independent comparison of date parts.

**Comment by bitwalker** [comment]
DateTime comparison (and the same for Date) works by converting to seconds since the Gregorian epoch and then comparing by granularity. Do you have a specific example where this is broken? If you are seeing incorrect diffs, then there is a bug for sure.

**Comment by brado** [comment]
I spent some more time testing this, and it turns out what appeared as an issue with the date comparison was actually an unexpected artifact of having functions called in separate processes asynchronously. It was a very cagey problem to get to the bottom of, but eventually I did. It was a little difficult to product a very small block of code to demonstrate this, so I didn't post it here. But suffice to say that Timex is working as expected. Thank you @bitwalker for your reply -- even that brief explanation helped me debug the problem. Closing this issue.
